providers/etcd/etcd.go

```diff
@@ -0,0 +1,92 @@
+package etcd
+
+import (
+	"time"
+	"errors"
+	"context"
+
+	clientv3 "go.etcd.io/etcd/client/v3"
+)
+
+type Config struct {
+	// etcd endpoints
+	Endpoints []string
+
+	// timeout
+	DialTimeout time.Duration
+
+	// prefix request option
+	Prefix bool
+
+	// limit request option
+	Limit bool
+
+	// number of limited pairs
+	nLimit int64
+
+	// key, key with prefix, etc.
+	Keypath string
```

Comment:
We should call this `Key` to be consistent with the etcd client language. https://pkg.go.dev/go.etcd.io/etcd/clientv3#OpGet